shuup/core/basket/objects.py

```diff
@@ -178,6 +178,8 @@ def clear_all(self):
         self._data = {}
         self.uncache()
         self.dirty = True
+        self.shipping_method = None
```

Comment:
Maybe you should set `self.shipping_method_id` to None instead of method. Doing this saves the None to the data.